detectors/resources/build.gradle

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2021 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+description = 'Google Cloud resource provider for OpenTelemetry'
+
+dependencies {
+	implementation(libraries.opentelemetry_api)
+	implementation(libraries.opentelemetry_sdk)
+	implementation(libraries.google_cloud_trace)
```

Comment:
I'm still not convinced that the Resource detector needs a dependency on Google Cloud Trace API.  It especially needs to not be true if we want to contribute this upstream into open telemetry.  Can you comment on why this is needed or what error you see removing it?

Reply:
It needs a dependency on Google Guava for com.google.common.base.MoreObjects.firstNonNull. I can instead add this to the root build and pass it down? Would that be a better alternative?

Reply:
Could we do the thing that `firstNonNull` does, manually? Code reuse is good, but a low-level library like OT (and its associated ecosystem) is better off avoiding dependencies. For example, we'd want to avoid the scenario where a consumer of OT gets their Guava dependency bumped to a new incompatible version, which breaks their code.

Reply:
Yep correct @punya @jsuereth Dependency removed